packages/supi/src/install/index.ts

```diff
@@ -211,33 +214,43 @@ export async function mutateModules (
       if (needsFullResolution) {
         throw new PnpmError('FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE', 'Cannot perform a frozen installation because the lockfile needs updates')
       }
-      if (!ctx.existsWantedLockfile) {
+      if (!ctx.existsWantedLockfile && !maybeOpts.ignorePackageManifest) {
         if (ctx.projects.some((project) => pkgHasDependencies(project.manifest))) {
           throw new Error(`Headless installation requires a ${WANTED_LOCKFILE} file`)
         }
+        return projects
+      }
+      const allImporterIds = maybeOpts.ignorePackageManifest
+
+      if (maybeOpts.ignorePackageManifest) {
+        logger.info({ message: 'Importing packages to virtual store', prefix: opts.lockfileDir })
       } else {
         logger.info({ message: 'Lockfile is up-to-date, resolution step is skipped', prefix: opts.lockfileDir })
-        try {
-          await headless({
-            currentEngine: {
-              nodeVersion: opts.nodeVersion,
-              pnpmVersion: opts.packageManager.name === 'pnpm' ? opts.packageManager.version : '',
-            },
-            currentLockfile: ctx.currentLockfile,
-            enablePnp: opts.enablePnp,
-            engineStrict: opts.engineStrict,
-            extraBinPaths: opts.extraBinPaths,
-            force: opts.force,
-            hoistedDependencies: ctx.hoistedDependencies,
-            hoistPattern: ctx.hoistPattern,
-            ignoreScripts: opts.ignoreScripts,
-            include: opts.include,
-            lockfileDir: ctx.lockfileDir,
-            modulesDir: opts.modulesDir,
-            ownLifecycleHooksStdio: opts.ownLifecycleHooksStdio,
-            packageManager: opts.packageManager,
-            pendingBuilds: ctx.pendingBuilds,
-            projects: ctx.projects as Array<{
+      }
+      try {
+        await headless({
+          allImporterIds,
+          currentEngine: {
+            nodeVersion: opts.nodeVersion,
+            pnpmVersion: opts.packageManager.name === 'pnpm' ? opts.packageManager.version : '',
+          },
+          currentLockfile: ctx.currentLockfile,
+          enablePnp: opts.enablePnp,
+          engineStrict: opts.engineStrict,
+          extraBinPaths: opts.extraBinPaths,
+          force: opts.force,
+          hoistedDependencies: ctx.hoistedDependencies,
+          hoistPattern: ctx.hoistPattern,
+          ignoreScripts: opts.ignoreScripts,
+          include: opts.include,
+          lockfileDir: ctx.lockfileDir,
+          modulesDir: opts.modulesDir,
+          ownLifecycleHooksStdio: opts.ownLifecycleHooksStdio,
+          packageManager: opts.packageManager,
+          pendingBuilds: ctx.pendingBuilds,
+          projects: maybeOpts.ignorePackageManifest
+            ? []
```

Comment:
so you're passing in an empty array here, that's why you need the additional property.

might be better to leave it as it was and just pass in the new `ignorePackageManifest` property instead. allImporterIds will not be needed in that case.